src/metadata/metadata.rs

```diff
@@ -48,11 +48,11 @@ impl Metadata {
         match key.as_str() {
             "Abstract" => {
                 let val = parse::parse_vec(val);
-                self.abs.join(Some(val));
+                self.abs.join(&val);
             }
             "Canonical Url" => {
                 let val = val.to_owned();
-                self.canonical_url.join(Some(val));
+                self.canonical_url.join(&Some(val));
```

Comment:
I still think it'd be better to make this just `self.canonical_url = Some(val);` and so on. No point in cloning the URLs and such, but...

Reply:
That's true. I would remove `Joinable` later.